File: app/src/api/product-catalog/data-transfer-objects/pad-color.ts
import { Brand, PadCategory, PadSeries, Stars } from '@/api';
import { PadFinish } from './pad-finish';
import { PadCut } from './pad-cut';
import { Rating } from './rating';
import { Image } from './image';
import { PadOption } from './pad-option';
import { PadMaterial } from './pad-material';
import { PadTexture } from './pad-texture';

export class PadColor {
    get label() {
        return `${this.series.brand.name} ${this.series.name} ${this.name}`
    }

    get imageUrl() {
        return this.image != null ? `${process.env.VUE_APP_API_DOMAIN}/image/${this.image}` : null;
    }

    get thumbnailUrl() {
        return this.image != null ? `${process.env.VUE_APP_API_DOMAIN}/image/${this.image}/thumb` : null;
    }

    constructor(
        public id: string,
        public series: PadSeries, // Used for quick lookup of series info only
        public name: string,
        public category: PadCategory,
        public material: PadMaterial,
        public texture: PadTexture,
        public cut: PadCut | null, // Comes from review
        public finish: PadFinish | null, // Comes from review
        public rating: Rating,
        public image: string | null = null,
        public options: PadOption[] = []
    ) { }
}

File: api/src/Domain/ProductCatalog/Entities/PadColor.cs
using System;
using System.Collections.Generic;
using DetailingArsenal.Domain.Shared;

namespace DetailingArsenal.Domain.ProductCatalog {
    public class PadColor : Entity<PadColor> {
        public string Name { get; set; }
        public PadCategory Category { get; set; }
        public PadMaterial Material { get; set; }
        public PadTexture Texture { get; set; }
        public ProcessedImage? Image { get; set; }
        public List<PadOption> Options { get; set; }

        public PadColor(string name, PadCategory category, PadMaterial material, PadTexture texture, ProcessedImage? image = null, List<PadOption>? options = null) {
            Id = Guid.NewGuid();
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            Image = image;
            Options = options ?? new();
        }

        public PadColor(Guid id, string name, PadCategory category, PadMaterial material, PadTexture texture, ProcessedImage? image = null, List<PadOption>? options = null) {
            Id = id;
            Name = name;
            Category = category;
            Material = material;
            Texture = texture;
            Image = image;
            Options = options ?? new();
        }
    }
}